components/src/AppModalLogin/appmodallogin.main.tsx

```diff
@@ -116,35 +123,46 @@ class AppModalLoginMain extends React.Component<AppModalLoginMainProps, AppModal
     handleModalClose();
   }
 
-  loginRegisteredUser(event) {
+  async loginRegisteredUser(event) {
+    event.preventDefault();
     const { username, password } = this.state;
     const { handleModalClose, onLogin, disableLogin } = this.props;
     this.setState({ isLoading: true });
     if (!disableLogin) {
-      if (localStorage.getItem(`${Config.cortexApi.scope}_oAuthRole`) === 'PUBLIC') {
-        loginRegistered(username, password).then((resStatus) => {
-          if (resStatus === 401) {
+      try {
+        if (localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`) != null) {
+          const result = await this.client.serverFetch('/oauth2/tokens', {
+            method: 'post',
+            useAuth: false,
+            urlEncoded: true,
+            body: {
+              username,
+              password,
+              grant_type: 'password',
+              role: 'REGISTERED',
+              scope: Config.cortexApi.scope,
+            },
+          });
+          localStorage.setItem(`${Config.cortexApi.scope}_oAuthToken`, `Bearer ${result.parsedJson.access_token}`);
+          localStorage.setItem(`${Config.cortexApi.scope}_oAuthRole`, result.parsedJson.role);
```

Comment:
You have the same piece of code again here. This should be refactored into a separate function.